lib/clipboard.zsh

```diff
@@ -67,20 +47,55 @@ function clipcopy() {
 #
 #   # Paste to a file
 #   clippaste > file.txt
-function clippaste() {
+#
+function detect-clipboard() {
   emulate -L zsh
-  if [[ $OSTYPE == darwin* ]]; then
-    pbpaste
-  elif [[ $OSTYPE == cygwin* ]]; then
-    cat /dev/clipboard
+
+  if [[ "${OSTYPE}" == darwin* ]] && (( ${+commands[pbcopy]} )) && (( ${+commands[pbpaste]} )); then
+    function clipcopy() { pbcopy < "${1:-/dev/stdin}"; }
+    function clippaste() { pbpaste; }
+  elif [[ "${OSTYPE}" == cygwin* ]]; then
+    function clipcopy() { cat "${1:-/dev/stdin}" > /dev/clipboard; }
+    function clippaste() { cat /dev/clipboard; }
+  elif [ -n "${WAYLAND_DISPLAY:-}" ] && (( ${+commands[wl-copy]} )) && (( ${+commands[wl-paste]} )); then
+    function clipcopy() { wl-copy < "${1:-/dev/stdin}"; }
+    function clippaste() { wl-paste; }
+  elif [ -n "${DISPLAY:-}" ] && (( ${+commands[xclip]} )); then
+    function clipcopy() { xclip -in -selection clipboard < "${1:-/dev/stdin}"; }
+    function clippaste() { xclip -out -selection clipboard; }
+  elif [ -n "${DISPLAY:-}" ] && $(( ${+commands[xsel]} )); then
+    function clipcopy() { xsel --clipboard --input < "${1:-/dev/stdin}"; }
+    function clippaste() { xsel --clipboard --output; }
+  elif (( ${+commands[lemonade]} )); then
+    function clipcopy() { lemonade copy < "${1:-/dev/stdin}"; }
+    function clippaste() { lemonade paste; }
+  elif (( ${+commands[doitclient]} )); then
+    function clipcopy() { doitclient wclip < "${1:-/dev/stdin}"; }
+    function clippaste() { doitclient wclip -r; }
+  elif (( ${+commands[win32yank]} )); then
+    function clipcopy() { win32yank -i < "${1:-/dev/stdin}"; }
+    function clippaste() { win32yank -o; }
+  elif [ -n "${TMUX:-}" ] && (( ${+commands[tmux]} )); then
+    function clipcopy() { tmux load-buffer "${1:-/dev/stdin}"; }
+    function clippaste() { tmux save-buffer -; }
   else
-    if (( $+commands[xclip] )); then
-      xclip -out -selection clipboard
-    elif (( $+commands[xsel] )); then
-      xsel --clipboard --output
-    else
-      print "clipcopy: Platform $OSTYPE not supported or xclip/xsel not installed" >&2
-      return 1
-    fi
+    function _retry_clipboard_detection_or_fail() {
+      local clipcmd="${1}"; shift
+      if detect-clipboard; then
+        "${clipcmd}" "$@"
+      else
+        print "${clipcmd}: Platform $OSTYPE not supported or xclip/xsel not installed" >&2
+        return 1
+      fi
+    }
+    function clipcopy() { _retry_clipboard_detection_or_fail clipcopy "$@"; }
+    function cilppaste() { _retry_clipboard_detection_or_fail clippaste "$@"; }
```

Comment:
Typo here

Reply:
Good eye! Thanks. Fixed, and verified no similar typos.